File: core/feature09a.go
package main

import (
	"math"
)

func feature09A(dd, aa, ll, tt float64) (ff1, ff2, hh, ss float64, p []float64, pp1ll1, pp1ll2, pp2ll1, pp2ll2 float64) {
	Rad := math.Pi / 180.0
	uu := 30.0
	if dd > 300.0 {
		uu = 40.0
	}
	r := dd / 2.0
	ee := math.Acos((r-aa)/r) / Rad
	gg := r * math.Sin(ee*Rad)
	pp := math.Pi / 180.0
	ii := tt / 2.0
	bb := r * ee * pp
	ff1 = gg
	ff2 = 2.0 * ff1
	hh = math.Sqrt(math.Pow(ll, 2.0) + math.Pow(aa, 2.0))
	ss = bb / ii
	p = make([]float64, 0)
	for xx := float64(0); xx <= tt/2.0; xx += 1.0 {
		// ff := xx * (360.0 / tt)
		qq := math.Sqrt(math.Pow(r*math.Cos(ee*Rad)-r*math.Cos(xx*(ee/ii)*Rad), 2.0) + math.Pow(r*math.Sin(ee*Rad)-r*math.Sin(xx*(ee/ii)*Rad), 2.0) + math.Pow(ll, 2.0))
		p = append(p, qq)
	}
	vv := 2.0 * math.Pi * r
	ww := uu / 2.0
	pp1ll1 = (1.0/4.0)*vv - bb + ww
	pp1ll2 = (3.0/4.0)*vv - bb + ww
	pp2ll1 = (1.0/2.0)*vv - bb + ww
	pp2ll2 = (1.0/2.0)*vv - bb + ww
	return
}

File: core/feature09a.js
function feature09a() {
  var parameters = extractParameters(["D", "A", "L", "T"]);
  return feature09a_core(parameters);
}

function feature09a_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var D = getParameter(parameters, "D");
  var A = getParameter(parameters, "A");
  var L = getParameter(parameters, "L");
  var T = getParameter(parameters, "T");
  var result = {};
  var Rad = Math.PI / 180.0;
  var U = 30.0;
  if (D > 300.0) {
    U = 40.0;
  }
  var r = D / 2.0;
  var E = Math.acos((r - A) / r) / Rad;
  var G = r * Math.sin(E * Rad);
  var P = Math.PI / 180.0;
  var I = T / 2.0;
  var B = r * E * P;
  result["F1"] = G;
  result["F2"] = 2.0 * G;
  result["H"] = Math.sqrt(Math.pow(L, 2.0) + Math.pow(A, 2.0));
  result["S"] = B / I;
  var p = [];
  for (var X = 0; X <= T / 2.0; X += 1.0) {
    // var F = X * (360.0 / T);
    var Q = Math.sqrt(Math.pow(r * Math.cos(E * Rad) - r * Math.cos(X * (E / I) * Rad), 2.0) + Math.pow(r * Math.sin(E * Rad) - r * Math.sin(X * (E / I) * Rad), 2.0) + Math.pow(L, 2.0));
    p.push(Q);
  }
  result["Points"] = p;
  var V = 2.0 * Math.PI * r;
  var W = U / 2.0;
  result["P1L"] = [
    (1.0 / 4.0) * V - B + W,
    (3.0 / 4.0) * V - B + W,
  ];
  result["P2L"] = [
    (1.0 / 2.0) * V - B + W,
    (1.0 / 2.0) * V - B + W,
  ];
  return result;
}
